class-amp-img.php

```diff
@@ -29,12 +29,13 @@ public function convert( $amp_attributes = array() ) {
 
 				// Workaround for https://github.com/Automattic/amp-wp/issues/20
 				// responsive + float don't mix
-				if ( isset( $attributes['class'] )
-					&& (
-						false !== strpos( $attributes['class'], 'alignleft' )
-						|| false !== strpos( $attributes['class'], 'alignright' )
-					)
-				) {
+				$img_class = isset( $attributes['class'] ) ? $attributes['class'] : '';
+				// alignleft or alignright
+				$pattern = "/alignleft|alignright/";
+				preg_match_all( $pattern, $img_class, $matches );
+				// test if there are any match
+				$is_aligned = count( $matches ) >= 1 && count( $matches[0] ) > 1;
+				if ( $is_aligned ) {
```

Comment:
I agree that the `if` isn't ideal, but there's small performance tradeoff with running `preg_match_all` every time even when the class attribute is not set.